### PR TITLE
opencc: use system rapidjson to fix gcc 14 error

### PR DESCRIPTION
**Pull request by jian-lin**
## Description of changes

The bundled rapidjson is not new enough and causes compilation
error[1] in gcc 14.

[1]: https://hydra.nixos.org/build/273250179/log

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/text/opencc/default.nix [hunk 1]
```diff
@@ -27,6 +29,19 @@ stdenv.mkDerivation rec {
       opencc # opencc_dict
     ];
 
+  buildInputs = [
+    rapidjson
+  ];
+
+  # TODO use more system dependencies
+  cmakeFlags = [
+    "-DUSE_SYSTEM_RAPIDJSON=ON"
```
Comment (NickCao) [inline pkgs/tools/text/opencc/default.nix:38]:
```suggestion
    (lib.cmakeBool "USE_SYSTEM_RAPIDJSON" true)
```

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 344004` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>goldendict-ng</li>
    <li>kdePackages.fcitx5-chinese-addons</li>
    <li>libsForQt5.fcitx5-chinese-addons</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>brise</li>
    <li>emacsPackages.rime</li>
    <li>fcitx5-rime</li>
    <li>ibus-engines.libpinyin</li>
    <li>ibus-engines.rime</li>
    <li>librime</li>
    <li>opencc</li>
    <li>rime-data</li>
  </ul>
</details>

**Comment by jian-lin** [comment]
> 3 packages failed to build:
> 
>     * goldendict-ng
> 
>     * kdePackages.fcitx5-chinese-addons
> 
>     * libsForQt5.fcitx5-chinese-addons

These failures presumably are caused by the failed nodejs: https://hydra.nixos.org/build/273514660.

**Comment by jian-lin** [comment]
I did some basic test:

```console
> ./result/bin/opencc --version
Open Chinese Convert (OpenCC) Command Line Tool
Version: 1.1.8

> echo "圖像" | ./result/bin/opencc -c s2twp.json
影像
```

**Pull request merged**